### ISSUE TITLE
[REVIEW]: PypeIt: The Python Spectroscopic Data Reduction Pipeline 

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2308]
**Submitting author:** @profxj (<a href="http://orcid.org/0000-0002-7738-6875">J. Xavier Prochaska</a>)
**Repository:** <a href="https://github.com/pypeit/PypeIt" target ="_blank">https://github.com/pypeit/PypeIt</a>
**Version:** v1.3
**Editor:** @dfm
**Reviewer:** @Gabriel-p, @astrom-tom
**Archive:** <a href="https://doi.org/10.5281/zenodo.4323006" target="_blank">10.5281/zenodo.4323006</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/a1a7029fb25f3915592f152154462f97/status.svg)](https://joss.theoj.org/papers/a1a7029fb25f3915592f152154462f97)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/a1a7029fb25f3915592f152154462f97"><img src="https://joss.theoj.org/papers/a1a7029fb25f3915592f152154462f97/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/a1a7029fb25f3915592f152154462f97/status.svg)](https://joss.theoj.org/papers/a1a7029fb25f3915592f152154462f97)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/pypeit/PypeIt" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@Gabriel-p & @astrom-tom, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @harpolea know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pypeit/PypeIt">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@profxj) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/pypeit/PypeIt">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@profxj) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @Gabriel-p, @astrom-tom it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2308 with the following error: 

 Can't find any papers to compile :-(

**Comment by harpolea** [comment]
@whedon generate pdf from branch joss-paper-branch

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper-branch. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02308/joss.02308/10.21105.joss.02308.pdf)

**Comment by kthyng** [comment]
Hi @Gabriel-p, @astrom-tom! This is just a friendly reminder about this JOSS review. Will you be able to work on this in the next week or so?

**Comment by Gabriel-p** [comment]
Hi @kthyng, I'll try to get to it this weekend

**Comment by astrom-tom** [comment]
I will do it this week.

**Comment by astrom-tom** [comment]
Hi @harpolea 
I am reviewing this software, it seems that there is no paper.md in the repository. Is that normal?

**Comment by harpolea** [comment]
Hi @astrom-tom – sometimes authors choose to put their paper in a different branch of the repository so as not to clutter the main repository. For this repo, the paper is in the `joss-paper-branch` branch: https://github.com/pypeit/PypeIt/tree/joss-paper-branch/paper

**Comment by astrom-tom** [comment]
ooh alright, thanks!

**Comment by astrom-tom** [comment]
@harpolea ,

Sorry to bug you, this is the first time I am reviewing for JOSS.
- For each identified problem/remarks I need to open an issue on the repo right?
- Once I am done with the review, what should I do?

**Comment by harpolea** [comment]
@astrom-tom that's totally fine – feel free to ask about anything you're unsure of!

Yes, if you identify any problems or have any suggestions, then please open an issue in the original repo. It's normally useful for the repo maintainers if you include 'JOSS' in the issue title so they can easily identify what the issue relates to, but obviously different organisations handle this in different ways, so go with whatever the repo's local style is! Also, if you include this issue URL somewhere in the issues you open, then the issue should appear in this thread as being 'mentioned'. 

Once you are finished with your review, it would be helpful if you can give a summary here. If you have ticked off everything in the checklist and don't think there's anything that needs to be changed or added, then you can simply say that you've looked over everything and recommend it for being accepted. If you have opened some issues and feel some changes should be made before the submission is accepted, then please give a brief overview of what changes you'd like to see (feel free to link to any issues you've opened, but if you've linked them as I suggested above then that won't necessary as they will already appear in this thread).

**Comment by astrom-tom** [comment]
Dear @kthyng, @harpolea 

I went throught the software and documentation, you can find my review below. Please let know if there is something missing. I am going to open issue for each of the 4 points highlighted below:

**1-Paper**
The paper is nicely written. I have a single remark about it: it could be useful to mention that pypeit is a set of commands design to perform the reduction (i.e. no need to know how to code in Python to use it).

**2-Installation**
I could install it with  the git clone section of the installation page. Although, I got an error:

```
Traceback (most recent call last):
  File "setup.py", line 12, in <module>
    from extension_helpers import get_extensions
ModuleNotFoundError: No module named 'extension_helpers'
```

After installing it manually, the python setup.py develop --user worked. **This should be added to the setup.py** of the package so it is installed automatically. The pip install from pypi crashed.

**3-Documentation:**
I believe a substantial work could be done about the documentation. I believe the authors could take example from existing documentation (e.g. https://fpavogt.github.io/fcmaker/index.html). My general impression is that the documentation is massive and that it is very hard to navigate in it. The documentation aims at providing help to people who want to use your code. For the moment it is not the case. I provide some details below where some improvements could be implemented.

_-Front page_
    
- I believe the front page could start with a quick overview of the code, with a list of instruments that are supported. It could be a quick summary of the paper.
- A link back to the repository should be added as well.
- A list of contributors could be added, as well a part about 'contributing' to the software. 
- A 'change log' section can also be added.
- The quick link sections of the front page (Running PypeIt, Data Products, Calibrations, etc...) can safely be removed. You can make a page for each section and link if you want from the front page. But just providing the list of links is not helpful at all. For example, the section 'spectrograph' links to multiple pages. Why not making a single page about spectrograph with each instrument as sections? This would make the documentation more compact and clearer (and would remove a lot of pages from the doctree, see below).
- Why is VLT-XShooter/Gemini GMOS in the Orphaned docs? Should be in the spectrograph documentation maybe?

_-About the Menu (left black):_
    
- The shortcut Menu on the left could be largely cleaned up. As most pages are links from PypeIT Cookbook pages, I don't think it is necessary to add them in the Menu (it makes it very confusing). Here should just go the general sections, but not all individual pages. 
    
 -   An optimal doctree could be :
        - Home (back to front page)
        - Installing PypeIt
        - PypeIt Cookbook
        - Spectrograph
        - Data Products
        - Calibrations
        - Master Frames (Although this could be part of Calibrations)
        - Reduction Algorithm
        - Miscellaneous (Documentation)
        - For developers (including testing)

-Code of conduct could go at the bottom of the doctree as it is intended firstly to developers.
 
_-Functionality documentation:_
    The 'pypeit package' page, with the list of modules, returns empty pages (just the name of the modules as page titles). The same goes for the 'PypeIt API' page. Why did you separate 'PypeIt API' from 'PypeIt Modules'?
    

_-Installation page:_ 
    At the bottom, in the blue note, what is the difference between 'all' and 'kast' options for tests?
    Please state it in that note.
    
_-Example usage:_
It would be nice, for the user (and also for the purpose of this review), to **add a 'step-by-step' page,** where you explicit everything from A-to-Z with a set of data that you provide and the expected results. **I will wait to have something like in order to comment on software functionality.**

**4-Testing:**

What is the difference between [2-Run the PypeIt unit tests] and [3-Try the test suite]? It should be clearly stated. For the moment I ran only the [2-Run the PypeIt unit tests] but I would like to know if I need to run the [3-Try the test suite]?.

I tried to run the unit tests with 'python setup.py tests'. It complained about missing numba, pytest, ginga and linetools. **This should be installed automatically during the pypeIt installation, please add it to the dependencies in the documentation AND in the setup.py.**

Results: -----144 passed, 89 skipped, 536 warnings in 33.60s------

Then I cloned the dev suite. The tests gave me the following error: 

```
Traceback (most recent call last):
  File "./pypeit_test", line 772, in <module>
    raise EnvironmentError('Enviroment must include PYPEIT_DEV, which points to the '
OSError: Enviroment must include PYPEIT_DEV, which points to the top-level directory with the dev-suite raw data.
```

Should this be taken care of with the installation? It is only after that I realised that the full instruction for running the dev suite is back on github! Please create a page with testing instruction on the documentation (see above). Downloading the data is not simple: The author proposes to use Google File stream which is not available to me ('To use Drive File Stream, you need to have a Google Account through work or school'). I will wait to have a clarification on the difference between [2-Run the PypeIt unit tests] and [3-Try the test suite] before running these tests.

**Comment by Gabriel-p** [comment]
@astrom-tom I see you marked **Community guidelines** as checked. Where did you find this in the docs? I was not able to find such requirements.

**Comment by astrom-tom** [comment]
It is in the readme of the repo. I asked them to add it in the front page.

**Comment by profxj** [comment]
@kthyng -- Kindly ping the reviewers again.  
Ball has been in their court for several weeks.

Thanks!

**Comment by kthyng** [comment]
@profxj I see a long review from @astrom-tom above but no response from you. Have you been working on edits and/or a response to that?

@Gabriel-p Will you be able to work on your review soon? 

@profxj Your editor is @harpolea, so please reach out to @harpolea first for your questions.

**Comment by profxj** [comment]
Got it.  @harpolea -- @Gabriel-p and @astrom-tom kindly broke their
review into a series of separate Issues on our GitHub:

https://github.com/pypeit/PypeIt/issues/993
https://github.com/pypeit/PypeIt/issues/994
https://github.com/pypeit/PypeIt/issues/995
https://github.com/pypeit/PypeIt/issues/1019

**Comment by kthyng** [comment]
Excellent! Bookkeeping is definitely an important part of the review process, to keep the issues linked together. Please add in links in all of the related software issues back to this review issue for tracking.

**Comment by Gabriel-p** [comment]
I'll resume today. Sorry for the delay.

**Comment by Gabriel-p** [comment]
I'm almost done with the review. The biggest hurdle now is that I do not have any input data files to test the package's functionality. I'm afraid that even if I did, I do not have the resources to do it:

> ## Grab a buff computer
> We recommend one with at least 32Gb of RAM and this might not be enough for many-detector instruments (e.g. Keck DEIMOS).
> 
> Multi-processors are good too, although only a portion of the code runs in parallel.

This prevents me from checking the **Functionality** point. @harpolea what should be done in this case?

@profxj  some minor issues with the article:

1. It mentions release 1.0 of the package, and v1.1 is already out.
2. There is I believe a missing reference (page 2):
    > PypeIt jointly performs the object extraction and b-spline @kelson2003 sky subtraction.

   Either that or I do not understand what that sentence means.

3. Perhaps a link to the "SPectral Image Typing (SPIT)" tool could be added?
4. Regarding the point "State of the field: Do the authors describe how this software compares to other commonly-used packages?", there are no other packages mentioned but perhaps this is because there are no similar packages available?

**Comment by profxj** [comment]
@Gabriel-p -- Thanks for the suggestions to the paper. 
I will make those edits.

As for a test run, here are a set of files for Shane Kastb
from our Development Suite:  https://drive.google.com/drive/folders/11ItdTKbY5c7Hfx-AnhSCvKcLW1g_mLAx?usp=sharing

These are small enough, 16Gb of RAM should be plenty.

**Comment by profxj** [comment]
Ok, addressed except the last one.  For SPIT, we reference it.
I attach the currently generated PDF in case the reviewers
are unable to make it themselves.

[klidmv2olhyhnshtsbjw.pdf](https://github.com/openjournals/joss-reviews/files/5277260/klidmv2olhyhnshtsbjw.pdf)

**Comment by Gabriel-p** [comment]
> 16Gb of RAM should be plenty

Still a no go (I have an old 8 Gb system, sorry). Perhaps @astrom-tom is able to perform a test run using these files?

> PypeIt jointly performs the object extraction and b-spline @kelson2003 sky subtraction

This is still in the PDF. Is it a missing reference or is 'b-spline @kelson2003 sky subtraction' a method? If so, please re-phrase or expand.

**Comment by profxj** [comment]
Have now fixed kelson on `joss-paper-branch`

**Comment by profxj** [comment]
Pinging @harpolea for an update.  
Much appreciated.

**Comment by Gabriel-p** [comment]
I'm pretty much done with the review (could I have a link to the latest PDF? @profxj), except that I can not check the **Functionality** item as the requirements are too much for my system to handle. @harpolea not sure what to do in these cases.

**Comment by profxj** [comment]
Attached @Gabriel-p 

[uhjqdqe28osajhr5bqzs.pdf](https://github.com/openjournals/joss-reviews/files/5397988/uhjqdqe28osajhr5bqzs.pdf)

**Comment by profxj** [comment]
@whedon - Raising this back up to your level.
Am hoping to publish this in 2020, if feasible.
Thanks.

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by profxj** [comment]
@kthyng -- Raising this back up to your level.
Am hoping to publish this in 2020, if feasible.
Thanks.

**Comment by dfm** [comment]
@profxj et al., I'm going pick up this review as editor to help @harpolea out!

**Comment by dfm** [comment]
@whedon assign @dfm as editor

**Comment by dfm** [comment]
:wave: @astrom-tom - it looks like there are a few issues on https://github.com/pypeit/PypeIt waiting for responses from you. Can you give us an idea of when you'll get a chance to get to those? Let me know if you have any questions!

**Comment by dfm** [comment]
Hi folks, I've sent @astrom-tom an email to see if we can get an update that way. I'll keep you all posted!

**Comment by dfm** [comment]
@Gabriel-p: I hadn't seen your remaining issue before - sorry about that!

For the functionality point: I agree that it would be good for the introductory tutorials to be self-contained and executable with moderate resources. @profxj: do you think that it would be feasible to provide a sample data set that could be tested with fewer resources? If that's not possible, let me know and we can figure out what to do. Thanks!

**Comment by profxj** [comment]
These Shane Kast blue data in the Drive below should work
with modest resources (<16Gb RAM).  But since I don't have
any such machine, I can't guarantee it.

https://drive.google.com/drive/folders/11ItdTKbY5c7Hfx-AnhSCvKcLW1g_mLAx?usp=sharing

**Comment by profxj** [comment]
These are the data from the StepbyStep Example:

https://pypeit.readthedocs.io/en/latest/step-by-step.html

**Comment by Gabriel-p** [comment]
I was able to successfully run the Step by step example with the files provided by @profxj on a system with 8 Gb of memory.

Small fix, in the **Spec2d** sections the commands:

    pypeit_show_2dspec Science/spec2d_b27-J1217p3905_KASTb_2015May20T045733.560.fits
    pypeit_show_1dspec Science/spec1d_b27-J1217p3905_KASTb_2015May20T045733.560.fits

should be written as `may` instead of `May`.

I'm thus checking the **Functionality** box, which completes my review.

**Comment by dfm** [comment]
Quick update here: Since I haven't heard back from @astrom-tom yet, I'm in the process of looking for a replacement reviewer. Thanks for your patience, all!

**Comment by profxj** [comment]
Hmm..  His review was quite good and we made quite a number
of changes to improve things (e.g. major overhaul of the docs).   

Can you explain why we need yet a 3rd reviewer now?

**Comment by dfm** [comment]
@profxj: I understand your frustration and I will certainly do everything I can to make everything go smoothly going forward, but we require two completed reviews for publication and if I can't get @astrom-tom to respond and finish the checklist then we're going to have to find another reviewer. Thanks for your understanding and patience!

**Comment by profxj** [comment]
Ok.  Kindly give the new reviewer a sense of the history, etc.
if you can.   Thanks.

**Comment by dfm** [comment]
Absolutely! Thanks again.

**Comment by astrom-tom** [comment]
Dear all,

I am terribly sorry for the time it took me with this review. I think I can finish it in the next couple of days. Let me know if that is still fine and I will put it as my priority to finish the review.

Best,

**Comment by profxj** [comment]
Works for me.  Your input thus far
has been very helpful.

**Comment by profxj** [comment]
ps.  note that the `joss-paper-branch` has evolved considerably
since you last touched it..

**Comment by profxj** [comment]
also, I just turned on that branch on `readthedocs`

should show up here soon:  https://pypeit.readthedocs.io/en/joss-paper-branch/

in case you want to see the bleeding edge.

**Comment by dfm** [comment]
@astrom-tom: Welcome back! This sounds great and thanks for your work so far.

**Comment by dfm** [comment]
@whedon generate pdf from branch joss-paper-branch

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper-branch. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: :point_left:

**Comment by astrom-tom** [comment]
Dear @profxj , @dfm 

I created a last issue [https://github.com/pypeit/PypeIt/issues/1082](url) with a couple of minor points. All the rest is fine for me.

**Comment by astrom-tom** [comment]
Dear @dfm 
All the comments have been implemented, so for me the paper is good to go!
(sorry again for the delay).

**Comment by dfm** [comment]
First, @Gabriel-p, @astrom-tom: thanks a lot for your constructive reviews!!

Next, @profxj: Things are looking good here! Here are the steps I need you to do before we head to final processing:

1. Are you planning on merging the `joss-paper-branch` into `master`? That's not required, but it might be a bit confusing if there is an out of date copy of the manuscript on `master`. Let me know how you want to handle this!
2. Once you've decided on the above, bump the version number and make a tagged release. That release should be archived somewhere like Zenodo and **the title and author list of the archive must _exactly_ match the paper**. Then, report the new version number and the DOI for the code archive back to this thread.
3. Finally, comment `@whedon generate pdf from joss-paper-branch` (or just `@whedon generate pdf` depending on what you do for step 1 above) and double check that you're happy with the article proofs. In particular, please check that author names and affiliations because that can be harder to edit later.

After those steps I think we're set!

**Comment by profxj** [comment]
Hi @dfm 

1.  Yes.  I foresee that happening in ~2-3 weeks when we release v1.3
2.  I presume then I should wait until we release v1.3 for this.  Right?

**Comment by profxj** [comment]
@whedon generate pdf from joss-paper-branch

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: :point_left:

**Comment by dfm** [comment]
@profxj: Sorry - I forgot the word "branch" in the whedon command. It should be:

`@whedon generate pdf from branch joss-paper-branch`

**Comment by profxj** [comment]
3.  The PDF looks great!

**Comment by dfm** [comment]
@profxj: Yes - the release should include all the changes associated with the review so getting those merged and included first would be good. So just ping me when you have that 1.3 release archived and I'll take it from there. Thanks!

**Comment by profxj** [comment]
@dfm -- v1.3 is on `master` now.  
Please proceed with publication.

Thanks!

**Comment by dfm** [comment]
@profxj: Great! What is the DOI for the archive of this release, described in step 2 in https://github.com/openjournals/joss-reviews/issues/2308#issuecomment-729782700 above?

**Comment by profxj** [comment]
How is this?



https://doi.org/10.5281/zenodo.4323006

**Comment by dfm** [comment]
The metadata *must be the same as the manuscript*:

- The title should be `PypeIt: The Python Spectroscopic Data Reduction Pipeline`
- And the author list (names and order) must also match

You can edit this fields on Zenodo.

**Comment by profxj** [comment]
How's this?



https://doi.org/10.5281/zenodo.4323006

**Comment by dfm** [comment]
Just about there! It looks like Frederick Davies is missing from the author list on Zenodo. Ping me when you've got that synchronized and we're good to go!

**Comment by profxj** [comment]
Ok, @dfm -- I think I got it.
Thanks for your help and patience.

**Comment by dfm** [comment]
@profxj: Awesome. Looks good now - thanks!

**Comment by dfm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: :point_left:

**Comment by dfm** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1086/648597 is OK
- 10.1086/683015 is OK
- 10.3847/1538-4357/aad18a is OK
- 10.3847/1538-4357/aae081 is OK
- 10.3847/1538-4357/aba52e is OK
- 10.1088/1538-3873/aace98 is OK
- 10.1086/375502 is OK
- 10.3847/1538-4357/ab8c45 is OK
- 10.3847/2041-8213/ab9c26 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by dfm** [comment]
@whedon set 10.5281/zenodo.4323006 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4323006" target="_blank">10.5281/zenodo.4323006</a> is the archive.

**Comment by dfm** [comment]
@whedon set v1.3 as version

**Comment by whedon** [comment]
OK. v1.3 is the version.

**Comment by dfm** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by dfm** [comment]
@profxj: I've handed this off to the Editors-in-Chief who will handle the final processing. They might have some last comments and then we'll get this published shortly.

Thanks again to @Gabriel-p and @astrom-tom for reviewing!

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1086/648597 is OK
- 10.1086/683015 is OK
- 10.3847/1538-4357/aad18a is OK
- 10.3847/1538-4357/aae081 is OK
- 10.3847/1538-4357/aba52e is OK
- 10.1088/1538-3873/aace98 is OK
- 10.1086/375502 is OK
- 10.3847/1538-4357/ab8c45 is OK
- 10.3847/2041-8213/ab9c26 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1989

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1989, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Hi @profxj, I am taking over to finish publishing your work. I read through the paper and I found a number of edits to make. I edited the paper and created a [PR](https://github.com/pypeit/PypeIt/pull/1111) you could consider merging for the corrections. Let me know what you decide.

**Comment by profxj** [comment]
approved!  Thanks @kthyng

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
Turns out there was another reference to the old version. Here is a new PR: https://github.com/pypeit/PypeIt/pull/1112

**Comment by profxj** [comment]
merged!  

thanks, @kthyng

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02308/joss.02308/10.21105.joss.02308.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1339286014989377537) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1990
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02308](https://doi.org/10.21105/joss.02308)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on your new publication @profxj! Thanks to editor @dfm and reviewers @Gabriel-p and @astrom-tom for your time and expertise!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02308/status.svg)](https://doi.org/10.21105/joss.02308)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02308">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02308/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02308/status.svg
   :target: https://doi.org/10.21105/joss.02308
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02308/status.svg)](https://doi.org/10.21105/joss.02308)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
